When files are removed from the directory during uploads, the app completely breaks.

Since the replay file statuses are kept in a database, if you remove the replay files between the time that it was added as an upload task and the time it was successfully uploaded, you will get constantly get *Connection Error*.

Example of the log:
```
17:28:39.761 [pool-2-thread-1          ] INFO  n.e.h.concurrent.tasks.UploadTask - Uploading replay Garden of Terror (5).StormReplay
17:28:39.761 [JavaFX Application Thread] WARN  n.e.h.models.ReplayFile - Infernal Shrines (3).StormReplay has no statuses.
17:28:39.762 [JavaFX Application Thread] ERROR n.e.h.services.UploaderService - UploadTask failed.
ninja.eivind.mpq.models.MpqException: Could not parse file
	at ninja.eivind.stormparser.StormParser.parseReplay(StormParser.java:33) ~[storm-parser-0.1.1.jar:na]
	at ninja.eivind.hotsreplayuploader.concurrent.tasks.UploadTask.lambda$call$49(UploadTask.java:48) ~[HotS%20Replay%20Uploader-jfx.jar:2.0.3]
	at java.util.Collections$SingletonList.forEach(Collections.java:4822) ~[na:1.8.0_60]
	at ninja.eivind.hotsreplayuploader.concurrent.tasks.UploadTask.call(UploadTask.java:45) ~[HotS%20Replay%20Uploader-jfx.jar:2.0.3]
	at ninja.eivind.hotsreplayuploader.concurrent.tasks.UploadTask.call(UploadTask.java:32) ~[HotS%20Replay%20Uploader-jfx.jar:2.0.3]
	at javafx.concurrent.Task$TaskCallable.call(Task.java:1423) ~[jfxrt.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_60]
	at javafx.concurrent.Service.lambda$null$493(Service.java:725) ~[jfxrt.jar:na]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_60]
	at javafx.concurrent.Service.lambda$executeTask$494(Service.java:724) ~[jfxrt.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_60]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_60]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_60]
Caused by: java.io.FileNotFoundException: /Users/moox/Library/Application Support/Blizzard/Heroes of the Storm/Accounts/56880185/1-Hero-1-5311518/Replays/Multiplayer/Garden of Terror (5).StormReplay (No such file or directory)
	at java.io.FileInputStream.open0(Native Method) ~[na:1.8.0_60]
	at java.io.FileInputStream.open(FileInputStream.java:195) ~[na:1.8.0_60]
	at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[na:1.8.0_60]
	at ninja.eivind.mpq.MpqArchive.getFiles(MpqArchive.java:53) ~[mpq-exploder-0.1.1.jar:na]
	at ninja.eivind.stormparser.StormParser.parseReplay(StormParser.java:29) ~[storm-parser-0.1.1.jar:na]
	... 12 common frames omitted
```

Deleting the database file fixes the issue.

--- a/src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java
+++ b/src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java
@@ -53,9 +53,12 @@ public class UploadTask extends Task<ReplayFile> {
                 preStatus = provider.getPreStatus(replay);
             } catch (MpqException e) {
                 LOG.warn("Could not parse replay, deferring to upload." + take, e);
-                preStatus = Status.NEW;
+
+                preStatus = Status.EXCEPTION;
             }
-            if (preStatus == Status.UPLOADED || preStatus == Status.UNSUPPORTED_GAME_MODE) {
+            if (preStatus == Status.UPLOADED ||
+                preStatus == Status.UNSUPPORTED_GAME_MODE ||
+                preStatus == Status.EXCEPTION) {
                 LOG.info("Parsed preStatus reported no need to upload "
                         + take.getFile() + " for provider " + provider.getName());
                 applyStatus(provider, preStatus);

--- a/src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java
+++ b/src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java
@@ -56,9 +56,7 @@ public class UploadTask extends Task<ReplayFile> {
 
                 preStatus = Status.EXCEPTION;
             }
-            if (preStatus == Status.UPLOADED ||
-                preStatus == Status.UNSUPPORTED_GAME_MODE ||
-                preStatus == Status.EXCEPTION) {
+            if (preStatus == Status.UPLOADED || preStatus == Status.UNSUPPORTED_GAME_MODE) {
                 LOG.info("Parsed preStatus reported no need to upload "
                         + take.getFile() + " for provider " + provider.getName());
                 applyStatus(provider, preStatus);

--- a/src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java
+++ b/src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java
@@ -19,6 +19,7 @@ import ninja.eivind.hotsreplayuploader.models.ReplayFile;
 import ninja.eivind.hotsreplayuploader.models.Status;
 import ninja.eivind.hotsreplayuploader.models.UploadStatus;
 import ninja.eivind.hotsreplayuploader.providers.Provider;
+import ninja.eivind.mpq.models.MpqException;
 import ninja.eivind.stormparser.StormParser;
 import ninja.eivind.stormparser.models.Replay;
 import org.slf4j.Logger;
@@ -45,8 +46,15 @@ public class UploadTask extends Task<ReplayFile> {
         providers.forEach(provider -> {
 
             final StormParser parser = new StormParser(take.getFile());
-            final Replay replay = parser.parseReplay();
-            final Status preStatus = provider.getPreStatus(replay);
+            Replay replay = null;
+            Status preStatus;
+            try {
+                replay = parser.parseReplay();
+                preStatus = provider.getPreStatus(replay);
+            } catch (MpqException e) {
+                LOG.warn("Could not parse replay, deferring to upload." + take, e);
+                preStatus = Status.NEW;
+            }
             if (preStatus == Status.UPLOADED || preStatus == Status.UNSUPPORTED_GAME_MODE) {
                 LOG.info("Parsed preStatus reported no need to upload "
                         + take.getFile() + " for provider " + provider.getName());

--- a/src/main/java/ninja/eivind/hotsreplayuploader/providers/hotslogs/HotsLogsProvider.java
+++ b/src/main/java/ninja/eivind/hotsreplayuploader/providers/hotslogs/HotsLogsProvider.java
@@ -69,6 +69,9 @@ public class HotsLogsProvider extends Provider {
         }
 
         final File file = replayFile.getFile();
+        if(!(file.exists() && file.canRead())) {
+            return Status.EXCEPTION;
+        }
 
         final String fileName = UUID.randomUUID() + ".StormReplay";
         LOG.info("Assigning remote file name " + fileName + " to " + replayFile);

--- a/src/main/java/ninja/eivind/hotsreplayuploader/services/UploaderService.java
+++ b/src/main/java/ninja/eivind/hotsreplayuploader/services/UploaderService.java
@@ -125,6 +125,11 @@ public class UploaderService extends ScheduledService<ReplayFile> implements Ini
         }
         try {
             final ReplayFile take = uploadQueue.take();
+            if(!take.getFile().exists()) {
+                fileRepository.deleteReplay(take);
+                files.remove(take);
+                return createTask();
+            }
             final UploadTask uploadTask = new UploadTask(providerRepository.getAll(), take);
             final Status oldStatus = take.getStatus();
 

